consumer/consumererror/throttle.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+	"time"
+)
+
+// TODO: Clean this by forcing all exporters to return an internal error type that always include the information about retries.
+
+// throttled wraps an error to indicate that it should be retried after a delay
+type throttled struct {
+	err   error
+	delay time.Duration
+}
+
+func (t throttled) Error() string {
+	return "Throttle (" + t.delay.String() + "), error: " + t.err.Error()
+}
+
+func (t throttled) Unwrap() error {
+	return t.err
+}
+
+// ThrottleRetryDelay returns the duration of time we should wait to retry, or
+// zero if the error is not a throttled retry error
+func ThrottleRetryDelay(err error) time.Duration {
```

Comment:
From @bogdandrutu in https://github.com/open-telemetry/opentelemetry-collector/pull/5551:

> Why this instead of exposing "Throttle" type and add a func on it RetryDelay?

Reply:
Why do you need to change the constructor?

Reply:
I was assuming you wanted to add additional data to the error, but I can't think of any examples where that would be useful :).

Reply:
I am waiting for you to do the change, unless I misunderstand something here.

Reply:
Ah, OK.  I'll make the change.  I thought you were just interested in why I had deviated from the previous implementation.  I can go back to that.

Reply:
Sorry for the delay.  Done.